### ISSUE TITLE
RFDETR pretrained model outputs COCO-91 label IDs instead of contiguous 0-79 indices

### ISSUE THREAD

**Issue by omkar-334** [open-edge-platform/training_extensions#6023]
RF-DETR's pretrained COCO weights use a 91-class detection head (COCO's original sparse category IDs). When `detector.model.reinitialize_detection_head(num_classes)` truncates the head to 80 classes, the output label indices still correspond to positions in the original 91-class space. Without remapping,  theclass predictions are silently wrong.   

RFDETR `coco_classes.py` - https://github.com/roboflow/rf-detr/blob/a81e659118f02c6d14b80c61b26d4baca34ca508/src/rfdetr/assets/coco_classes.py#L89                                                                                                        

Before:  
<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/6eee0b64-c508-4786-af1b-0d0b4b7db8eb" />  
                                
After:  
<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/780f2177-0466-4d56-979b-a6cae3d402fc" />

**Comment by omkar-334** [comment]
Script to reproduce  
```python
"""Detect on a single frame and save annotated image with boxes, labels, and scores."""

import cv2
import numpy as np
import torch

from otx.backend.native.models.detection.rfdetr import RFDETR
from otx.data.entity import OTXSampleBatch
from otx.data.entity.base import ImageInfo

COCO_CLASSES = [
    "person", "bicycle", "car", "motorcycle", "airplane", "bus", "train", "truck",
    "boat", "traffic light", "fire hydrant", "stop sign", "parking meter", "bench",
    "bird", "cat", "dog", "horse", "sheep", "cow", "elephant", "bear", "zebra",
    "giraffe", "backpack", "umbrella", "handbag", "tie", "suitcase", "frisbee",
    "skis", "snowboard", "sports ball", "kite", "baseball bat", "baseball glove",
    "skateboard", "surfboard", "tennis racket", "bottle", "wine glass", "cup",
    "fork", "knife", "spoon", "bowl", "banana", "apple", "sandwich", "orange",
    "broccoli", "carrot", "hot dog", "pizza", "donut", "cake", "chair", "couch",
    "potted plant", "bed", "dining table", "toilet", "tv", "laptop", "mouse",
    "remote", "keyboard", "cell phone", "microwave", "oven", "toaster", "sink",
    "refrigerator", "book", "clock", "vase", "scissors", "teddy bear", "hair drier",
    "toothbrush",
]
CONF = 0.5
DEVICE = "mps" if torch.backends.mps.is_available() else ("cuda" if torch.cuda.is_available() else "cpu")

model = RFDETR(label_info=80, model_name="rfdetr_base")
model.eval()
model.to(DEVICE)

cap = cv2.VideoCapture("videos/bikes-1.mp4")
for _ in range(60):
    ret, frame = cap.read()
cap.release()

# Inline preprocess_frame (video.py doesn't exist on develop)
inp_h, inp_w = model.data_input_params.input_size
mean = model.data_input_params.mean
std = model.data_input_params.std
ori_h, ori_w = frame.shape[:2]

rgb = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
resized = cv2.resize(rgb, (inp_w, inp_h))
tensor = torch.from_numpy(resized).permute(2, 0, 1).float()
if all(m < 1.0 for m in mean):
    tensor = tensor / 255.0
tensor = (tensor - torch.tensor(mean).view(3, 1, 1)) / torch.tensor(std).view(3, 1, 1)

img_info = ImageInfo(
    img_idx=0,
    img_shape=(inp_h, inp_w),
    ori_shape=(ori_h, ori_w),
    scale_factor=(inp_h / ori_h, inp_w / ori_w),
)
batch = OTXSampleBatch(images=tensor.unsqueeze(0).to(DEVICE), imgs_info=[img_info])

with torch.no_grad():
    preds = model.predict_step(batch, batch_idx=0)

bboxes = preds.bboxes[0].cpu().numpy()
scores = preds.scores[0].cpu().numpy()
labels = preds.labels[0].cpu().numpy()

# Draw boxes with class names and scores
for i in range(len(bboxes)):
    if scores[i] < CONF:
        continue
    x1, y1, x2, y2 = map(int, bboxes[i])
    label_idx = int(labels[i])
    name = COCO_CLASSES[label_idx] if label_idx < len(COCO_CLASSES) else f"cls{label_idx}"
    text = f"{name} {scores[i]:.2f}"

    cv2.rectangle(frame, (x1, y1), (x2, y2), (0, 255, 0), 2)
    cv2.putText(frame, text, (x1, y1 - 8), cv2.FONT_HERSHEY_SIMPLEX, 0.6, (0, 255, 0), 2)

cv2.imwrite("issue_video.png", frame)
print(f"Saved issue_video.png")
for i in range(len(bboxes)):
    if scores[i] < CONF:
        label_idx = int(labels[i])
        name = COCO_CLASSES[label_idx] if label_idx < len(COCO_CLASSES) else f"cls{label_idx}"
        print(f"  {name}: {scores[i]:.3f}")
```

**Comment by omkar-334** [comment]
This is how I fixed this for my demo branch ( branched off from `develop`) - 
Commit - https://github.com/open-edge-platform/training_extensions/commit/e9ff87941220ac13d005fbdf8a9cd96d18eef250
Demo (RFdetr for tracking) - https://colab.research.google.com/github/omkar-334/otx/blob/gsoc-tracker-new/demo_new.ipynb

**Comment by kprokofi** [comment]
Hi @omkar-334 
Did you try instantiate a model with 91 classes? RFDETR(label_info=91 ...)
Since this model was trained originally with 91 classes, I think we may use this intentionally

**Comment by omkar-334** [comment]
hey @kprokofi i did try to instantiate it with 91 classes. This works, but it has 2 caveats - 
1. The user will have to import rfdetr's classes for mapping 
```python
from rfdetr.util.coco_classes import COCO_CLASSES
```  
 2. The model outputs 0-indexed labels but COCO-91 IDs are 1-indexed, so the user still needs to apply a +1 offset before looking up class names.
```python
label_idx = int(labels[i]) + 1  # 0-indexed model output → 1-indexed COCO-91 ID
name = COCO_CLASSES.get(label_idx, f"cls{label_idx}")
```

**Comment by omkar-334** [comment]
Another side-effect is - 
RF-DETR's class indexes are non-contiguous - indexes `12, 26, 29, 30, 45, 66, 68, 69, 71, 83, 91` are missing....
In the end, the 91 classes come down to 80 since 11 are missing.  
Will this cause any issue in metrics, exported models or tracking?

**Comment by kprokofi** [comment]
Hi, this issue will be addressed after the 3.0 release. As I can see it is only related to usage of original pertained weights and doesn't affect fine-tuning, which is our primary goal
